perf/flame.jl

```diff
@@ -59,17 +60,17 @@ allocs = @allocated SciMLBase.step!(integrator)
 @info "`allocs ($job_id)`: $(allocs)"
 
 allocs_limit = Dict()
-allocs_limit["flame_perf_target"] = 4656
-allocs_limit["flame_perf_target_tracers"] = 204288
-allocs_limit["flame_perf_target_edmfx"] = 253440
-allocs_limit["flame_perf_diagnostics"] = 3016328
-allocs_limit["flame_perf_target_diagnostic_edmfx"] = 920960
+allocs_limit["flame_perf_target"] = 10_155_936
```

Comment:
I think this needs to be fixed

Reply:
When I first looked into this, I found that there were a lot of allocations in NCDatasets. I'll look again because this is probably not the case for this particular run (since we don't write to disk)

Reply:
Ah no, now I remember. The allocations come from these functions:
```
RRTMGPI.array2field(
            FT.(cache.radiation.radiation_model.face_clear_sw_flux_dn),
            axes(state.f),
        )
```

Reply:
Hm, I wonder if it's just the `FT` (https://github.com/CliMA/ClimaCore.jl/issues/1372), so maybe this can easily be fixed with `$(FT(...))`?

Reply:
The other allocating diagnostics have

```
out .= Geometry.UVector.(cache.precomputed.ᶜu).components.data.:1
```

or

```
out .= Geometry.VVector.(cache.precomputed.ᶜu).components.data.:1
```

How do I deal with this?

Reply:
We need a function like 
https://github.com/CliMA/ClimaCore.jl/blob/30a42ece976016c713c70a5e54bc3fb41568ef11/src/Geometry/conversions.jl#L175

Reply:
An easy workaround for now:
```julia
u_component(u::LocalVector) = u.u
v_component(u::LocalVector) = u.v
w_component(u::LocalVector) = u.w
out .= u_component.(Geometry.UVector.(cache.precomputed.ᶜu))
```

Reply:
This worked and dramatically reduced allocations! :)

Reply:
Among the remaining allocations I am finding:
```
julia> @allocated integrator.p
6128

julia> @allocated integrator.u
160

julia> @allocated integrator.t
128
```
Which I find quite interesting and I don't understand. 

I also added a check for excessive allocations in the constructor of the simulation.

The EDMF case still allocates a lot. The same diagnostics allocate much more compared to the non EDMF case. I am still debugging this.

Reply:
Here are where the EDMF allocations happen:

Function # 68 is the one at line 699 on the right panel

![image](https://github.com/CliMA/ClimaAtmos.jl/assets/9167485/a9194c6e-8ca6-46cd-9a74-e6ae81bfabcd)

Reply:
This could simply be dynamic dispatch? Not sure there is much you can do about this.